pkg/ruler/compat.go

```diff
@@ -39,9 +39,12 @@ func (a *appender) AddFast(l labels.Labels, ref uint64, t int64, v float64) erro
 }
 
 func (a *appender) Commit() error {
-	_, err := a.pusher.Push(user.InjectOrgID(context.Background(), a.userID), client.ToWriteRequest(a.labels, a.samples, nil, client.RULE))
+	wq := client.ToWriteRequest(a.labels, a.samples, nil, client.RULE)
+	_, err := a.pusher.Push(user.InjectOrgID(context.Background(), a.userID), wq)
+	client.ReuseSlice(wq.Timeseries)
```

Comment:
This doesn't seem to be part of the PR description?

Reply:
it's another bugfix.
I'll update changelog and PR title.

Reply:
I don't think this is correct. `a.pusher` is an instance of `Distributor` and the slice reusing is already done there. I think we should rollback this change.

Reply:
I've rolled back this change.